TweetMate/src/main/java/com/ybsystem/tweetmate/activities/PhotoActivity.java

```diff
@@ -0,0 +1,152 @@
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import androidx.appcompat.app.ActionBar;
+import androidx.fragment.app.FragmentManager;
+import androidx.viewpager.widget.ViewPager;
+
+import com.ybsystem.tweetmate.R;
+import com.ybsystem.tweetmate.adapters.pager.PhotoPagerAdapter;
+import com.ybsystem.tweetmate.fragments.dialog.ListDialog;
+import com.ybsystem.tweetmate.libs.photo.PhotoViewPager;
+import com.ybsystem.tweetmate.usecases.DownloadUseCase;
+import com.ybsystem.tweetmate.utils.MediaUtils;
+import com.ybsystem.tweetmate.utils.ToastUtils;
+import com.ybsystem.tweetmate.utils.StorageUtils;
+
+import java.util.ArrayList;
+
+import static com.ybsystem.tweetmate.resources.ResString.*;
+
+public class PhotoActivity extends ActivityBase {
+    // Pager
+    private ViewPager mPhotoPager;
+
+    // Intent
+    private String mType; // "PROFILE", "BANNER", "MEDIA"
+    private int mSize; // 0(:small), 1(:medium), 2(:large), 3(:orig)
+    private int mPagerPosition;
+    private ArrayList<String> mImageURLs;
+    private boolean mIsThirdMedia;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Set view
+        setContentView(R.layout.activity_photo);
+        mPhotoPager = (PhotoViewPager) findViewById(R.id.view_pager);
+        setContentView(mPhotoPager);
+
+        // Init
+        mType = getIntent().getStringExtra("TYPE");
+        mSize = getIntent().getIntExtra("SIZE", -1);
+        mPagerPosition = getIntent().getIntExtra("PAGER_POSITION", -1);
+        mImageURLs = (ArrayList<String>) getIntent().getSerializableExtra("IMAGE_URLS");
+        mIsThirdMedia = getIntent().getBooleanExtra("THIRD_MEDIA", false);
+
+        // Set pager
+        setActionBarTitle();
+        setPhotoPager();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        if (!mIsThirdMedia) {
+            getMenuInflater().inflate(R.menu.photo, menu);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            // 保存する
+            case R.id.item_save:
+                if (!StorageUtils.isPermitted(this)) {
+                    StorageUtils.requestPermission(this);
+                    return true;
+                }
+                showQualitySelection(true);
+                return true;
+            // 他サイズで表示
+            case R.id.item_resize:
+                showQualitySelection(false);
+                return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void setActionBarTitle() {
+        ActionBar bar = getSupportActionBar();
+        if (bar != null) {
+            int currentPage = mPagerPosition + 1;
+            int totalPage = mImageURLs.size();
+            bar.setTitle(String.format("%s (%d/%d)", STR_PHOTO, currentPage, totalPage));
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs32/TweetMate-for-Android/blob/0a5013e538213bacb2774f24bfa4ed6959fa720c/TweetMate/src/main/java/com/ybsystem/tweetmate/activities/PhotoActivity.java#L89"></span>Implicitly using the default locale is a common source of bugs: Use <code>String.format(Locale, ...)</code> instead</td>
    </tr>
  </tbody>
</table>